### ISSUE TITLE
In memory only analysis has regressed, need to migrate Dart-Services to new API

### ISSUE THREAD

**Issue by lukechurch** [dart-lang/sdk#29088]
This is a counterpart bug for https://github.com/dart-lang/dart-services/issues/286. @mit-mit requested a bug to be filed here so progress can be tracked as part for the release. This is currently blocking upgrading DartPad.

Summary:

- DartPad uses analyzer to analyze source that are in memory only, they should not materialised to disk
- The API for doing this appears to have regressed, @danrubel's recomendation is to migrate to the new mechanisms for handling files based on summaries
- There appears to be a difficulty in using the new API whilst loading the SDK files from disk, but the user supplied code from memory

To resolve this we need:
- A recommendation as to how to use the new API
- An integration test in the SDK to prevent this regressing

Many thanks

**Comment by bwilkerson** [comment]
@scheglov I'm guessing that we (also?) need to make it possible for the byte store to be kept in memory rather than on disk.

**Comment by devoncarew** [comment]
@lukechurch, can we instead change the dartpad back-end to write a file to disk? This would not then be blocked by analysis server changes, and would be a fairly small task IIRC.

**Comment by scheglov** [comment]
Do we have problems with memory-only files in Analysis Server?
The idea was that if you send `analysis.updateContent` with `AddContentOverlay`, you can "create" files in memory without writing anything to the file system.

**Comment by danrubel** [comment]
This issue [has been addressed for 1.23](https://github.com/dart-lang/dart-services/commits/analyzer_upgrade?after=a21ca2f6c94338f5a5f082663cab50e61cd257f1+0) and the analyzer_upgrade branch merged into master.

@lukechurch 
Please reopen if this issue crops up again.
